code/numerical_analysis/monte_carlo/src/integral_montecarlo.c

```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+/*Part of Cosmos by OpenGenus Foundation*/
+
+double
+integral_monte_carlo(double a, double b, double point_count, double (*f)(double))
+{
+    double res, x, y;
+    y = 0;
+    for (int i = 0; i < point_count; ++i)
+    {
+        x = ((b - a) * (double)random() / RAND_MAX) + a;
+        y += f(x);
+    }
+    res = (b - a) * (y / point_count);
+    return (res);
+}
+
+int
+main(void)
+{
+    double a, b, point_count, res;
+    printf("Enter interval: \n");
+    printf("a: \n");
+    scanf("%lf", &a);
+    printf("b: \n");
+    scanf("%lf", &b);
+    printf("Enter point count: ");
+    scanf("%lf", &point_count);
+    res = integral_monte_carlo(a, b, point_count, ({double f(double x){return x*x + x;} f;}));
+    printf("Result =  %lf \n", res);
```

Comment:
the variable res is pretty unnecessary.

Reply:
@arnavb What do you mean?

Reply:
You don't need the variable at all, instead just do:

```C
printf("Result =  %lf \n", integral_monte_carlo(a, b, point_count, ({double f(double x){return x*x + x;} f;})));
```
Just a recommendation.